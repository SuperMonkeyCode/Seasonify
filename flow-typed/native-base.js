/* @flow */
declare module 'native-base' {
  declare var exports: any;
}
