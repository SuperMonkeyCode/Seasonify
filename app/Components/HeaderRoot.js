import React from 'react';
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';

export default function HeaderRoot({
  title,
  navigation,
}: {
  title: string,
  navigation: Object,
}) {
  const openDrawer = () => {
    navigation.navigate('DrawerOpen');
  };
  return (
    <Header>
      <Left>
        <Button transparent onPress={openDrawer}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>{title}</Title>
      </Body>
      <Right />
    </Header>
  );
}
