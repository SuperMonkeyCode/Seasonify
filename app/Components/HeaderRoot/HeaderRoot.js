/* @flow */
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  title: string,
  navigation: Object,
  /* connect */
  /* state */
  /* handlers */
  handleDrawer: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    handleDrawer: ({ navigation }: Props) => () => {
      navigation.navigate('DrawerOpen');
    },
  }),
);

function HeaderRoot({ title, handleDrawer }: Props) {
  return (
    <Header>
      <Left>
        <Button transparent onPress={handleDrawer}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>{title}</Title>
      </Body>
      <Right />
    </Header>
  );
}

export default enhance(HeaderRoot);
