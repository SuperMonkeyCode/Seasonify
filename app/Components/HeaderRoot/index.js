/* @flow */
import HeaderRoot from './HeaderRoot';

export default HeaderRoot;
