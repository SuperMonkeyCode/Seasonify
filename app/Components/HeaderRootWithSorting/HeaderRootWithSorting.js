/* @flow */
import { Button, Text } from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

import AppConfig from 'Config/AppConfig';
import HeaderRoot from 'Components/HeaderRoot';

type Props = {
  /* parent */
  title: string,
  navigation: Object,
  sortingKey: SortingValue,
  /* connect */
  sorting: SortingValue,
  toggleSorting: Function,
  /* HOC */
  toggleAction: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    toggleAction: ({ toggleSorting, sortingKey }: Props) => () => {
      toggleSorting(sortingKey);
    },
  }),
);

function HeaderRootWithSorting({ title, navigation, toggleAction, sorting }: Props) {
  return (
    <HeaderRoot
      title={title}
      navigation={navigation}
      toggleButton={
        <Button transparent small onPress={toggleAction}>
          <Text>
            {AppConfig.textWhenSortingAlpha[sorting]}
          </Text>
        </Button>
      }
    />
  );
}

export default enhance(HeaderRootWithSorting);
