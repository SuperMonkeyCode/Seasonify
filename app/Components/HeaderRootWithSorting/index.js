/* @flow */
import { connect } from 'react-redux';

import { sortingActions, sortingSelectors } from 'State/sortingState';

import HeaderRootWithSorting from './HeaderRootWithSorting';

export default connect(
  (state, { sortingKey }) => ({
    sorting: sortingSelectors.getSorting(state, sortingKey),
  }),
  {
    toggleSorting: sortingActions.toggleSorting,
  },
)(HeaderRootWithSorting);
