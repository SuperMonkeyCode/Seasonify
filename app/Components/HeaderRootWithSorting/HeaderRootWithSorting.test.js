import { Button } from 'native-base';
import React from 'react';

import HeaderRoot from 'Components/HeaderRoot';

import HeaderRootWithSorting from './HeaderRootWithSorting';

function setup(specificProps = {}) {
  const props = {
    title: 'Title',
    navigation: NavigationMock,
    sortingKey: 'friend',
    sorting: 'DESC',
    toggleSorting: jest.fn(),
    ...specificProps,
  };
  const component = shallowDive(<HeaderRootWithSorting {...props} />, HeaderRoot);
  // console.warn(component.debug());
  return {
    component,
    props,
  };
}

describe('Rendering when default state', () => {
  const { component } = setup();
  it('should match', () => {
    expect(component.dive()).toMatchSnapshot();
  });
});

/* ========= Events & Functions ========= */

describe('Events & Functions', () => {
  const { component, props } = setup();

  describe('pressing on the sorting button', () => {
    component.props().toggleButton.props.onPress();

    it('calls toggleSorting', () => {
      expect(props.toggleSorting).toBeCalledWith(props.sortingKey);
    });
  });
});
