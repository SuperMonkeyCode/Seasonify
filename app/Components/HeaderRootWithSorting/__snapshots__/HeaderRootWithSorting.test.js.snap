// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering when default state should match 1`] = `
<withHandlers(HeaderRoot)
  navigation={
    Object {
      "goBack": [Function],
      "navigate": [Function],
      "setParams": [Function],
      "state": Object {
        "params": Object {},
      },
    }
  }
  title="Title"
  toggleButton={
    <Styled(Button)
      onPress={[Function]}
      small={true}
      transparent={true}
    >
      <Styled(Text)>
        Z-A
      </Styled(Text)>
    </Styled(Button)>
  }
/>
`;
